marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -36,14 +36,16 @@ function getTypesInfo() {
     for (const artifactType of Object.keys(registeredArtifactTypes)) {
         if (registeredArtifactTypes.hasOwnProperty(artifactType)) {
             const artifactLibrary = registeredArtifactTypes[artifactType];
+            const updateRoles = artifactLibrary.getPermissions().filter((perm) => perm.capability === 'update').map((perm) => perm.roleId);
+            const currentRoles = xdmp.getCurrentRoles().toArray();
```

Comment:
```suggestion
            const updateRoles = artifactLibrary.getPermissions().filter((perm) => perm.capability === 'update').map((perm) => String(perm.roleId));
            const currentRoles = xdmp.getCurrentRoles().toArray().map(String);
```

Reply:
Updated to map to strings so they're accurately compared.